tests/Phinx/Config/ConfigTest.php

```diff
@@ -95,10 +95,24 @@ public function testReturnsEmptyArrayWithEmptyDataDomain()
      * @covers \Phinx\Config\Config::getDefaultEnvironment
      */
     public function testGetDefaultEnvironmentUsingDatabaseKey()
+    {
+        $configArray = $this->getConfigArray();
+        $configArray['environments']['default_environment'] = 'production';
+        $config = new Config($configArray);
+        $this->assertEquals('production', $config->getDefaultEnvironment());
+    }
+
+    /**
+     * @covers \Phinx\Config\Config::getDefaultEnvironment
+     */
+    public function testDeprecatedDefaultDatabase()
     {
         $configArray = $this->getConfigArray();
         $configArray['environments']['default_database'] = 'production';
         $config = new Config($configArray);
+
+        $this->expectDeprecation();
+        $this->expectExceptionMessage('default_database in the config has been deprecated since 0.12, use default_environment instead.');
         $this->assertEquals('production', $config->getDefaultEnvironment());
```

Comment:
It looks like the deprecated warning in the code actually stops phpunit from continuing with testing, and so there's no way to validate that the function actually does return `production` it looks like.

Reply:
You can check how CakePHP using a function wrapper does it, this seems to work well so far.

Reply:
Ah, that makes sense. Added another testcase which does that wrapper to validate the value works as expected, as well as kept existing test to validate the deprecation notice is raised.

Reply:
@MasterOdin You just need to use the deprecated() wrapper here. I believe.

Reply:
There is no `deprecated` wrapper available within phinx codebase at the moment. Doing so would probably require adding a new dev dependency to some other cakephp package for this, which is probably not worth while for a singular test.